PlexRequests.UI/Modules/DonationLinkModule.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Threading.Tasks;
+
+using Nancy;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NLog;
+
+using PlexRequests.Core;
+using PlexRequests.Core.SettingModels;
+using PlexRequests.Helpers;
+using PlexRequests.UI.Models;
+
+namespace PlexRequests.UI.Modules
+{
+    public class DonationLinkModule : BaseAuthModule
+    {
+        public DonationLinkModule(ICacheProvider provider, ISettingsService<PlexRequestSettings> pr) : base("customDonation", pr)
+        {
+            Cache = provider;
+
+            Get["/", true] = async (x, ct) => await GetCustomDonationUrl(pr);
+        }
+
+        private ICacheProvider Cache { get; }
+
+        private static Logger Log = LogManager.GetCurrentClassLogger();
+
+        private async Task<Response> GetCustomDonationUrl(ISettingsService<PlexRequestSettings> pr)
+        {
+            PlexRequestSettings settings = pr.GetSettings();
+            try
+            {
+                if (settings.EnableCustomDonationUrl)
+                {
+                    JObject o = new JObject();
+                    o["url"] = "\""+ settings.CustomDonationUrl + "\"";
+                    o["message"] = "\"" + settings.CustomDonationMessage + "\"";
+                    return Response.AsJson(o);
```

Comment:
This I think would be a bit easier to read if it was a dynamic object? e.g.

```
return Response.AsJson(new { url = settings.CustomDonationUrl, message = settings.CustomDonationMessage});
```
Since `AsJson` will automatically serialise the object for you,